### PR TITLE
feat: Add generated Content-ID to changesets

### PR DESCRIPTION
**Pull request by techaware**
This PR adds generated content ids to the changeset parts in batch requests.
Closes [2255](https://github.com/SAP/cloud-sdk-js/issues/2255).

Closes SAP/cloud-sdk-backlog#622.

### CODE REVIEW

**Comment by FrankEssenberger** [comment]
Relates to BLI: https://github.com/SAP/cloud-sdk-backlog/issues/622

@techaware,

thanks for creating this PR to enable `Content-ID` support. We will do a review and try to include it quckly.
It was the first item in our input queue currently.


Best
Frank

**Comment by marikaner** [comment]
@techaware Please give us more insight in your use case and why it is necessary to pass a custom content-id (see [here](https://github.com/SAP/cloud-sdk-js/issues/2255#issuecomment-1087646903)). For now I would like to adjust your PR to just add a generated content-id from the SDK.

File: packages/odata-common/src/request-builder/batch/batch-request-serializer.ts [hunk 1]
```diff
@@ -57,14 +59,14 @@ export function serializeRequest(
     ([key, value]) => `${voca.titleCase(key)}: ${value}`
   );
 
+  const method = request.requestConfig.method.toUpperCase();
+
   return [
     'Content-Type: application/http',
     'Content-Transfer-Encoding: binary',
+    ...(method !== 'GET' ? [`Content-Id: ${uuid()}`] : []),
```
Comment (jjtang1985) [inline packages/odata-common/src/request-builder/batch/batch-request-serializer.ts:67]: content-id is used with change set, which cannot have get request 👍

**Review by jjtang1985 — APPROVED** [review]
LGTM for `content-id`

For 'Content-Type' and 'Cotent-Transfer-Encoding' mentioned in the original issue, we might need a follow up?

File: CHANGELOG.md [hunk 2]
```diff
@@ -24,6 +24,7 @@
 - [http-client] Remove `method` from `defaultDisallowedKeys` to not filter out custom http method when using `filterCustomRequestConfig`.
 - [odata-common] Support adding custom http method in `addCustomRequestConfiguration` to overwrite the default http method.
 - [openapi] Support adding custom http method in `addCustomRequestConfiguration` to overwrite the default http method.
+- [odata-common] Generate mandatory `Content-Id` headers for changeset parts in batch requests payloads (see [OData Spec](http://docs.oasis-open.org/odata/odata/v4.01/odata-v4.01-part1-protocol.html#sec_MultipartBatchRequestBody)).
```
Comment (jjtang1985) [inline CHANGELOG.md:27]: You have to merge the main branch, so this will be shown in the new release.

**Pull request merged**